### ISSUE TITLE
Function tryTarWithGzip (carvel-vendir/pkg/vendir/fetch/archive.go) loses link/symlink/mode/UID/GID/etc. in extracting .tar and .tar.gz files

### ISSUE THREAD

**Issue by yeahdongcn** [carvel-dev/vendir#174]
**What steps did you take:**
[A clear and concise description steps that can be used to reproduce the problem.]
Fetch a `tar` file through `HTTP` through `DirectoryContents` config, everything works as expected except the extracted file contents are different from `tar xvf <<path-to-tar-file>>`.

**What happened:**
[A small description of the issue]
`.tar` or `.tar.gz` file are extracted but the content is different than `tar xvf <<path-to-tar-file>>`.

**What did you expect:**
[A description of what was expected]
`.tar` or `.tar.gz` file are extracted and the content is the same as `tar xvf <<path-to-tar-file>>`.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

**Environment:**

- vendir version (execute `vendir --version`):
Head of `develop` branch.
- OS (e.g. from `/etc/os-release`):
```bash
❯ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by yeahdongcn** [comment]
Just checked the implementation, `tryTarWithGzip` only handles `tar.TypeReg`.

**Comment by cppforlife** [comment]
@yeahdongcn yeah this was meant to be an implementation that covers majority of use cases and remains simplest. i think we would be open to a PR to enhance it but i would probably ask for an opt-in to enable this feature (as some kind of a configuration option on http section)
